components/function-controller/internal/controllers/serverless/function_controller.go

```diff
@@ -0,0 +1,730 @@
+// +kubebuilder:rbac:groups="serverless.kyma-project.io",resources=functions/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups="serving.knative.dev",resources=services,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups="serving.knative.dev",resources=services/status,verbs=get
+// +kubebuilder:rbac:groups="batch",resources=jobs,verbs=get;list;watch;create;update;patch;delete;deletecollection
+// +kubebuilder:rbac:groups="batch",resources=jobs/status,verbs=get
+// +kubebuilder:rbac:groups="",resources=events,verbs=create;patch
+func (r *FunctionReconciler) Reconcile(request ctrl.Request) (ctrl.Result, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	instance := &serverlessv1alpha1.Function{}
+	err := r.Get(ctx, request.NamespacedName, instance)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	log := r.Log.WithValues("kind", instance.GetObjectKind().GroupVersionKind().Kind, "name", instance.GetName(), "namespace", instance.GetNamespace(), "version", instance.GetGeneration())
+
+	log.Info("Collecting ConfigMaps")
+	var configMaps corev1.ConfigMapList
+	if err := r.resourceClient.ListByLabel(ctx, instance.GetNamespace(), r.functionLabels(instance), &configMaps); err != nil {
+		log.Error(err, "Cannot list ConfigMaps")
+		return ctrl.Result{}, err
+	}
+	log.Info("Collecting Jobs")
+	var jobs batchv1.JobList
+	if err := r.resourceClient.ListByLabel(ctx, instance.GetNamespace(), r.functionLabels(instance), &jobs); err != nil {
+		log.Error(err, "Cannot list Jobs")
+		return ctrl.Result{}, err
+	}
+	log.Info("Collecting Service")
+	service := &servingv1.Service{}
+	if err := r.Client.Get(ctx, request.NamespacedName, service); err != nil {
+		if apierrors.IsNotFound(err) {
+			service = nil
+		} else {
+			log.Error(err, "Cannot get Service %s", instance.GetName())
+			return ctrl.Result{}, err
+		}
+	}
+
+	switch {
+	case r.isOnConfigMapChange(instance, configMaps.Items, service):
+		return r.onConfigMapChange(ctx, log, instance, configMaps.Items)
+	case r.isOnJobChange(instance, jobs.Items, service):
+		return r.onJobChange(ctx, log, instance, configMaps.Items[0].GetName(), jobs.Items)
+	default:
+		return r.onServiceChange(ctx, log, instance, service)
+	}
+}
+
+func (r *FunctionReconciler) isOnConfigMapChange(instance *serverlessv1alpha1.Function, configMaps []corev1.ConfigMap, service *servingv1.Service) bool {
+	image := r.buildExternalImageAddress(instance)
+	if service != nil && service.Spec.Template.Spec.Containers[0].Image == image {
+		return false
+	}
+
+	return len(configMaps) != 1 ||
+		instance.Spec.Source != configMaps[0].Data[configMapFunction] ||
+		r.sanitizeDependencies(instance.Spec.Deps) != configMaps[0].Data[configMapDeps] ||
+		"handler.main" != configMaps[0].Data[configMapHandler]
```

Comment:
![image](https://user-images.githubusercontent.com/17271979/79200615-fe6b4c80-7e36-11ea-9fcb-af8149283628.png)
 
:D

and besides, use `configMapHandler` variable, not a string